src/network/networkd-address.c

```diff
@@ -774,6 +774,48 @@ int config_parse_label(const char *unit,
         return 0;
 }
 
+int config_parse_lifetime(const char *unit,
+                                  const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+        Network *network = userdata;
+        _cleanup_address_free_ Address *n = NULL;
+        int r, k;
+
+        assert(filename);
+        assert(section);
+        assert(lvalue);
+        assert(rvalue);
+        assert(data);
+
+        r = address_new_static(network, section_line, &n);
+        if (r < 0)
+                return r;
+
+        if (!streq(rvalue, "forever") && !streq(rvalue, "infinity")) {
```

Comment:
This ain't gonna work. If `n` already existed,  `n->cinfo.ifa_prefered` is not reset properly.

Also, STR_IN_SET(rvalue, "forever", "infinity") seems nicer.